near-sdk-sim/src/runtime.rs

```diff
@@ -87,7 +87,7 @@ impl GenesisConfig {
 
 #[derive(Debug, Default, Clone)]
 pub struct Block {
-    prev_block: Option<Box<Block>>,
+    prev_block: Option<Arc<Block>>,
```

Comment:
Do we actually need `Arc` here? Would `Rc` suffice?

Reply:
Good question, I haven't actually thought about this. I'd say that:

* both would probably work
* it doesn't really matter which one we pick here (like, this worked before by just copying the whole thing)
* Arc seems to be a better default choice, as that preserves Sync-ness of the Block

Reply:
I wonder if we use `Box` recursively somewhere in nearcore as well

Reply:
@evgenykuzyakov nerdsniped! Here's the list of recursive types from nearcore: 

https://github.com/rust-analyzer/rust-analyzer/commit/ffd92831327b4cb9c19bc1fc440f2a0eb6842c89

Nothing there seems problematic. 

Though, the tool is rather simplistic, it might've misssed some